### PR TITLE
add support for ed25519

### PR DESCRIPTION
**Pull request by aclevername**
### Description
Closes https://github.com/weaveworks/eksctl/issues/4779

working on a PR to the `kops` `pki` library we use under the hood, as its the more appropriate place for this IMO. This can be used as a temporary workaround

I wanted to make sure the fingerprint value we produce matches whats displayed in ec2 console. We do this for example with existing rsa keys:
image
```
[ℹ]  using SSH public key "/home/jake/.ssh/id_rsa.pub" as "eksctl-jk-nodegroup-mng-1-9f:f0:b4:09:16:9f:f0:ab:43:4a:79:14:04:71:5b:7b"
```

Now works with the new keys:

image

```
[ℹ]  using SSH public key "/home/jake/weave/eksctl/key.pub" as "eksctl-jk-nodegroup-mng-1-HvE7+gmH78VS53+iPuRDh/gKjVo26OzYU/qOnJWAgyk"

### CODE REVIEW

File: pkg/ssh/client/ssh.go [hunk 1]
```diff
@@ -32,13 +33,11 @@ func LoadKeyFromFile(filePath, clusterName, ngName string, ec2API ec2iface.EC2AP
 	}
 
 	key := string(fileContent)
-
-	fingerprint, err := pki.ComputeAWSKeyFingerprint(key)
+	fingerprint, err := fingerprint(filePath, fileContent)
```
Comment (Skarlso) [inline pkg/ssh/client/ssh.go:36]: Interesting that this wasn't picked up by golangci-lint. I guess we have that check disabled.

You are not checking the `err` but using the `fingerprint` result immediately. So first, check the `err`. :)

Comment (aclevername) [reply pkg/ssh/client/ssh.go:35]: oh yeah. Fixed

**Review by Skarlso — APPROVED** [review]

File: pkg/ssh/client/ssh.go [hunk 2]
```diff
@@ -48,6 +48,24 @@ func LoadKeyFromFile(filePath, clusterName, ngName string, ec2API ec2iface.EC2AP
 	return keyName, nil
 }
 
+func fingerprint(filePath string, key []byte) (string, error) {
+	pk, _, _, _, err := ssh.ParseAuthorizedKey(key)
+	if err != nil {
+		return "", fmt.Errorf("parsing key %q: %w", filePath, err)
+	}
+
+	if strings.Contains(pk.Type(), "ed25519") {
```
Comment (Skarlso) [inline pkg/ssh/client/ssh.go:57]: Ah hang on. This could just be `pk.Type() == ssh.KeyAlgoED25519`.

Comment (aclevername) [reply pkg/ssh/client/ssh.go:57]: I'll make it more specific

**Pull request merged**